### PR TITLE
Updated Blimp Strat

### PR DESCRIPTION
**Pull request by Reguas**
Now the "Daveyy" cutscene triggers as player falls down. The cutscene will always contain Michael regardless of the current character, although when it ends, the player gains back control of a current character.

Also slightly changed the position of the blimp in order to make the fall more persistent.

### CODE REVIEW

**Comment by Last0xygen** [comment]
Why block the Player for an additional 10 seconds? 🤔

**Comment by Reguas** [comment]
Yeah, could make it skippable, but that would cause problems while in missions, showing "mission failed" screen while playing the cutscene. Plus, 13 seconds for an effect isn't _that_ much. Gravity Field and Autopilot, for example, last twice as long, and player also can do nothing.

**Comment by rollschuh2282** [comment]
there is a way to change a character during a cutscene with this native.
https://alloc8or.re/gta5/nativedb/?n=0xE40C1C56DF95C2E8
maybe a check could get implemented so the current player appears in the cutscene rather than always michael.

**Comment by Reguas** [comment]
Thank you, successfully changed this.

**Review by TheRealAyCe — COMMENTED** [review]
What are the hashes for? Shouldn't they be defined globally somewhere instead of copypasted into specific effects?

**Comment by Reguas** [comment]
Idk what were they doing there 😐

File: ChaosMod/Effects/db/Player/PlayerBlimpStrats.cpp [hunk 1]
```diff
@@ -1,24 +1,68 @@
...
+
+		REGISTER_ENTITY_FOR_CUTSCENE(player, "MICHAEL", 0, 0, 64);
+		
+		START_CUTSCENE(0);
+		WAIT(8500);
+		STOP_CUTSCENE_IMMEDIATELY();
+
+		REMOVE_CUTSCENE();
+
+		Ped pedDave = CreatePoolPed(4, daveHash, -442.2f, 1059.25f, 326.66f, 180.6f);
```
Comment (pongo1231) [inline ChaosMod/Effects/db/Player/PlayerBlimpStrats.cpp:61]: Would be better to just spawn a normal ped and use `SET_PED_AS_NO_LONGER_NEEDED` instead, no need to make Dave persistent

Comment (Reguas) [reply ChaosMod/Effects/db/Player/PlayerBlimpStrats.cpp:61]: Gotcha

**Comment by rollschuh2282** [comment]
not sure if needed, but maybe a check for the real mission would be smart.
bc doesn´t sound very good if the game tries to start the mission and then the script loads the cutscene as well.
sounds like it could break the mission script.

**Comment by Reguas** [comment]
> not sure if needed, but maybe a check for the real mission would be smart

No, that's fine. `IS_CUTSCENE_PLAYING()` is enough for the effect to work correctly

**Comment by OnlyRealNubs** [comment]
Maybe we should set the player invulnerable, in the possibly inevitable case that the player could die when they are jumping out.

**Comment by Reguas** [comment]
> If the player is already at low HP, and they hit the ground when it is deactivated, it could kill them.
Wdym, it's active unless the player is on the ground

**Comment by OnlyRealNubs** [comment]
When I tested, I would get damage from hitting the pavement.

**Comment by OnlyRealNubs** [comment]
> When I tested, I would get damage from hitting the pavement.

@Reguas After doing more testing. This only happens my fps is locked to 30, from the `Console Experience` effect.

**Comment by Reguas** [comment]
Kinda sus, but ok, moved invincibility deactivation after the cutscene

**Review by OnlyRealNubs — APPROVED** [review]

**Pull request merged**